src/lib.rs

```diff
@@ -597,14 +597,15 @@ use dtls::Fingerprint;
 use dtls::{Dtls, DtlsEvent};
 
 mod ice;
-use ice::IceAgent;
-use ice::IceAgentEvent;
-use ice::IceCreds;
-pub use ice::{Candidate, CandidateKind};
+pub use ice::{
+    Candidate, CandidateKind, IceAgent, IceAgentEvent, IceConnectionState, IceCreds, IceError,
+};
 
 mod io;
 use io::DatagramRecv;
 
+pub use io::{StunError, StunMessage};
+
```

Comment:
On the root level of str0m, I only want the types related to the "standard" WebRTC use case. This includes `Candidate` and `CandidateKind`, since a user must interact with those to create an `Rtc` instance. They should remain where they are.

However the rest of the ice types are a "low level use case", which I think should live in an `ice` module. A module we don't expose by default, but with certain feature flags (`ice`?).

Having them as a top level module API wise, doesn't mean we need to pub our internal `ice` mod. See below trick with `rtp_` and `#[path = …]`. I suggest we do this the same way.

I think it would also be fine to have `StunMessage` and `StunError` sitting under this new "synthehtic" `ice` module.

As a starting point we could keep `IceAgent` internally, but rename the export (`pub use ice_::IceAgent as Agent`) in this synthetic module. The drawback is that type name changes/aliases can be confusing in Rust. However the alternative of dropping the `Ice`-prefix would make for inconsistent internal naming in str0m (compare `IceError`, `RtpError`, `DtlsError` etc). We can revisit this later if it sucks too much.

Reply:
Sounds good to me!